exporter/splunkhecexporter/client.go

```diff
@@ -615,6 +619,73 @@ func (c *client) stop(context.Context) error {
 	return nil
 }
 
-func (c *client) start(context.Context, component.Host) (err error) {
+func (c *client) start(ctx context.Context, host component.Host) (err error) {
+
+	httpClient, err := buildHTTPClient(c.config, host, c.telemetrySettings)
+	if err != nil {
+		return err
+	}
+
+	if c.config.HecHealthCheckEnabled {
+		healthCheckURL, _ := c.config.getURL()
+		healthCheckURL.Path = c.config.HealthPath
+		if err := checkHecHealth(httpClient, healthCheckURL); err != nil {
+			return fmt.Errorf("health check failed: %w", err)
+		}
+	}
+	url, _ := c.config.getURL()
+	c.hecWorker = &defaultHecWorker{url, httpClient, buildHTTPHeaders(c.config, c.buildInfo)}
 	return nil
 }
+
+func checkHecHealth(client *http.Client, healthCheckURL *url.URL) error {
+
+	req, err := http.NewRequest("GET", healthCheckURL.String(), nil)
+	if err != nil {
+		return consumererror.NewPermanent(err)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	err = splunk.HandleHTTPCode(resp)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildHTTPClient(config *Config, host component.Host, telemetrySettings component.TelemetrySettings) (*http.Client, error) {
+	// we handle compression explicitly.
+	config.HTTPClientSettings.Compression = ""
```

Comment:
We shouldn't mutate the original config. We should make a copy instead. Can you please create an issue to follow up on this?

Reply:
ok. I think we do this in signalfx too. I'll make an issue. I guess those objects will eventually be immutable, is that right?

Reply:
Forcing them to be immutable would be pretty complicated, but as a rule, we've been trying to to not mutate the original user-supplied config

Reply:
OK, so would you agree a better fix is to validate the user doesn't set a compression algo then? That would take care of this, and might be more elegant.

Reply:
That would be better, but we also override MaxIdleConns and MaxIdleConnsPerHost